code/modules/paperwork/contract.dm

```diff
@@ -300,3 +300,6 @@
 	user.mutations.Add(XRAY)
 	user.mind.AddSpell(new /obj/effect/proc_holder/spell/targeted/view_range(null))
 	return ..()
+
+/obj/item/weapon/paper/contract/infernal/fire_act()
+	return  //Nope.
```

Comment:
If you want to prevent the contract from burning, you may also want to add `burn_state = LAVA_PROOF` to the infernal contract's list of variables